core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -108,7 +108,10 @@ public void initialize(String name, Map<String, String> properties) {
         "Cannot initialize HadoopCatalog because warehousePath must not be null or empty");
 
     this.catalogName = name;
-    this.warehouseLocation = LocationUtil.stripTrailingSlash(inputWarehouseLocation);
+    this.warehouseLocation = inputWarehouseLocation;
+    if (!warehouseLocation.endsWith("://")) {
```

Comment:
I think this should be rather fixed directly in `LocationUtil.stripTrailingSlash()` and we also need tests in `TestLocationUtil`

Reply:
Makes sense, as the util was also used for file paths I didn't put it in there. 
moved it into LocationUtil and added tests